packages/wonder-blocks-dropdown/components/dropdown.test.js

```diff
@@ -0,0 +1,158 @@
+                        icon={icons.search}
+                        aria-label="search"
+                    />
+                )}
+            </Dropdown>,
+        );
+    });
+
+    afterEach(() => {
+        unmountAll();
+    });
+
+    it("opens when the anchor is clicked", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+
+        // Act
+        anchor.simulate("click");
+
+        // Assert
+        expect(dropdown.state("open")).toBe(true);
+    });
+
+    it("can handle keyboard navigation", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+        anchor.simulate("click");
+        const dropdownCore = dropdown.find(DropdownCore);
+
+        // Act & Assert
+        dropdownCore.simulate("keydown", {keyCode: keyCodes.down});
+        dropdownCore.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdownCore.instance().focusedIndex).toBe(0);
+
+        dropdownCore.simulate("keydown", {keyCode: keyCodes.down});
+        dropdownCore.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdownCore.instance().focusedIndex).toBe(1);
+
+        dropdownCore.simulate("keydown", {keyCode: keyCodes.down});
+        dropdownCore.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdownCore.instance().focusedIndex).toBe(2);
+    });
+
+    it("closes itself on escape", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+        anchor.simulate("click");
+        const dropdownCore = dropdown.find(DropdownCore);
+
+        // Act
+        dropdownCore.simulate("keydown", {keyCode: keyCodes.escape});
+        dropdownCore.simulate("keyup", {keyCode: keyCodes.escape});
+
+        // Assert
+        expect(dropdown.state("open")).toBe(false);
+    });
+
+    it("closes itself on tab", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+        anchor.simulate("click");
+        const dropdownCore = dropdown.find(DropdownCore);
+
+        // Act
+        dropdownCore.simulate("keydown", {keyCode: keyCodes.tab});
+        dropdownCore.simulate("keyup", {keyCode: keyCodes.tab});
+
+        // Assert
+        expect(dropdown.state("open")).toBe(false);
+    });
+
+    it("closes itself on an external mouse click", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+        anchor.simulate("click");
+
+        // Act
+        document.dispatchEvent(new MouseEvent("mouseup"));
+
+        // Assert
+        expect(dropdown.state("open")).toBe(false);
+    });
+
+    it("open on down key when focused", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+        anchor.simulate("click");
+
+        // Act
+        anchor.simulate("keydown", {keyCode: keyCodes.down});
+        anchor.simulate("keyup", {keyCode: keyCodes.down});
+
+        // Assert
+        expect(dropdown.state("open")).toBe(true);
+    });
+});
```

Comment:
The tests look good.  There are a few hole in the the coverage though: https://codecov.io/gh/Khan/wonder-blocks/compare/b395cbd861e5211cd2f8133773ffe31f720991db...9311baab3540b1c865786c93848bf74ff0f6ba3a/diff.